05/app.js

```diff
@@ -1,53 +1,90 @@
 const stats = {
-    paragraphs: {
-        'p1': 0,
-    },
-    links: {
-        '/dolor.html': 0,
-    }
-};
+	paragraphs: {
+		p1: 0,
+	},
+	links: {
+		'/dolor.html': 0,
+	},
+}
 
 /* tutaj umieść swój kod */
+const pList = document.querySelectorAll('p')
 
+pList.forEach(function (item) {
+	item.addEventListener('click', function (e) {
+		e.preventDefault()
 
-/* nie modyfikuj kodu poniżej, ale przeanalizuj go */
+		if (e.target.getAttribute('href') === null) {
+			countParagraphs(e)
+		} else if (e.target.getAttribute('href') !== null) {
+			countLinks(e)
+		}
+	})
+})
 
-const statsElement = document.querySelector('.stats');
-const fireCustomEvent = function(element, name) {
-    console.log(element, '=>', name);
+function countParagraphs(e) {
+	const elId = e.currentTarget.dataset.id
+
+	if (stats.paragraphs[elId] !== undefined) {
+		stats.paragraphs[elId]++
+	} else {
+		stats.paragraphs[elId] = 0
+		stats.paragraphs[elId]++
+	}
```

Comment:
Zwróć uwagę, że w `countParagraphs` oraz  `countLinks` jest sporo podobnych działań.
Może dałoby się napisać 1 funkcję, która przyjmuje przez parametr informacje o tym jaką właściwość zaktualizować tj. `paragraphs` lub `links` oraz jaki "identyfikator" ma mieć aktualizowana wartość. Specjalnie piszę tak na około, aby nie zdradzić za wiele - traktuj to jako zadanie dodatkowe ;)

Reply:
Tak racja, poprawione. Czy coś mogę jeszcze usprawnić? :)